loader.go

```diff
@@ -43,98 +43,95 @@ type graphDescription struct {
 
 // ParseJSON converts a JSON network definition string into
 // a flow.Graph object that can be run or used in other networks
-func ParseJSON(js []byte) *Graph {
+func ParseJSON(js []byte, factory *Factory) *Graph {
 	// Parse JSON into Go struct
 	var descr graphDescription
 	err := json.Unmarshal(js, &descr)
 	if err != nil {
+		fmt.Println("Error parsing JSON", err)
```

Comment:
Please clean up all debug prints before merging, and make use of `error` to return errors to caller.